package.json

```diff
@@ -22,6 +22,7 @@
     "colorette": "^1.2.1",
     "electron-to-chromium": "^1.3.591",
     "escalade": "^3.1.1",
+    "js-yaml": "^3.14.0",
```

Comment:
It will increase `browserslist` dependencies by 3 dependencies (`js-yaml`, `argparse`, `esprima`). It is too much for `--update-db` which you need only one per month.

Can we do it without library?

Reply:
It's much easier to parse the pnpm lock file using a yaml parser, rather than with regular expressions. I changed the way that the pnpm lock file is updated to account for cases where more than one version of `caniuse-lite` is installed (delete nodes and let pnpm rewrite the lock file), and that would be a lot messier to do via naïve line-by-line string parsing. In the grand scheme of things, I think having code that is easier to read, update, and maintain trumps a small dependency list.

There are other yaml parsers that could be used:
* [yaml](https://www.npmjs.com/package/yaml) -- 0 deps!
* [yamljs](https://www.npmjs.com/package/yamljs) -- 2 deps

Reply:
`pnpm` use case is so rare. We can’t add one dependency just to simplify work for 1% of users. 99% of users will download an unnecessary package.

I think users' comfort is much more important than the developer’s comfort. Let’s use RegExp for now.

In the next major release of Browserslist, I will move `update-db` to a separated package and will clean up the code with Yaml parsers there.

Reply:
How many users even look at package dependencies, though? I certainly didn't when I started using browserslist -- I just followed the instructions at the postcss website which linked to the best practices section of the readme. If you are worried about having an extra install for some users, the yaml parser could be a peer or optional dependency that gets installed if it isn't present during the pnpm workflow.

Reply:
> How many users even look at package dependencies, though? I certainly didn't when I started using browserslist

The problem is big `node_modules` containing a lot of unnecessary code.

Keeping optional dependency will force us to show a good error message on the missed YAML parser. It is still extra work. It will be better to use regexp.

Reply:
I disagree. Hand-parsing is always more fragile.

I have rewritten the pnpm code and removed the yaml parser dependency.